pkg/controller/eventing/subscription/reconcile.go

```diff
@@ -0,0 +1,331 @@
+	if err != nil {
+		glog.Warningf("Feiled to unserialize Sinkable target: %s", err)
+		return "", err
+	}
+	if s.Status.Sinkable != nil {
+		return s.Status.Sinkable.DomainInternal, nil
+	}
+	return "", fmt.Errorf("status does not contain targetable")
+}
+
+// resolveFromChannelable fetches an object based on ObjectReference. It assumes that the
+// fetched object then implements Subscribable interface and returns the ObjectReference
+// representing the Channelable interface.
+func (r *reconciler) resolveFromChannelable(namespace string, ref *corev1.ObjectReference) (*duckv1alpha1.Subscription, error) {
+	obj, err := r.fetchObjectReference(namespace, ref)
+	if err != nil {
+		glog.Warningf("Failed to fetch From target %+v: %s", ref, err)
+		return nil, err
+	}
+
+	c := duckv1alpha1.Subscription{}
+	err = duck.FromUnstructured(*obj, &c)
+	return &c, err
+}
+
+// fetchObjectReference fetches an object based on ObjectReference.
+func (r *reconciler) fetchObjectReference(namespace string, ref *corev1.ObjectReference) (*unstructured.Unstructured, error) {
+	//	resourceClient, err := r.CreateResourceInterface2(r.restConfig, ref, namespace)
+	resourceClient, err := r.CreateResourceInterface(namespace, ref)
+	if err != nil {
+		glog.Warningf("failed to create dynamic client resource: %v", err)
+		return nil, err
+	}
+
+	return resourceClient.Get(ref.Name, metav1.GetOptions{})
+}
+
+func (r *reconciler) reconcileFromChannel(namespace string, subscribable corev1.ObjectReference, callDomain string, resultDomain string, deleted bool) error {
+	glog.Infof("Reconciling From Channel: %+v call: %q result %q deleted: %v", subscribable, callDomain, resultDomain, deleted)
+
+	s, err := r.fetchObjectReference(namespace, &subscribable)
+	if err != nil {
+		return err
+	}
+
+	c := duckv1alpha1.Channel{}
+	err = duck.FromUnstructured(*s, &c)
+	if err != nil {
+		return err
+	}
+
+	glog.Infof("Found From Channelable: %+v", c)
+
+	// Add our subscription to the object
+	after := c.DeepCopyObject().(*duckv1alpha1.Channel)
+	after.Spec.Channelable = &duckv1alpha1.Channelable{
+		Subscribers: []duckv1alpha1.ChannelSubscriberSpec{{CallableDomain: "foobar"}},
+	}
+
+	patch, err := createPatch(c, after)
+	if err != nil {
+		return err
+	}
+	glog.Infof("PATCH is: %+v", patch)
+
+	patchBytes, err := patch[0].MarshalJSON()
+	if err != nil {
+		glog.Warningf("failed to marshal json patch: %s", err)
+		return err
+	}
+
+	resourceClient, err := r.CreateResourceInterface(namespace, &subscribable)
+	if err != nil {
+		glog.Warningf("failed to create dynamic client resource: %v", err)
+		return err
+	}
+	patched, err := resourceClient.Patch(c.Name, types.JSONPatchType, patchBytes)
+	if err != nil {
+		glog.Warningf("Failed to patch the object: %s", err)
+		return err
+	}
+	glog.Warningf("Patched resource: %+v", patched)
+	return nil
+}
+
+func (r *reconciler) CreateResourceInterface(namespace string, ref *corev1.ObjectReference) (dynamic.ResourceInterface, error) {
```

Comment:
Not sure if this is useful, or even how it works, but controller-runtime just gained the ability to work with unstructured objects: https://github.com/kubernetes-sigs/controller-runtime/pull/101

Reply:
Awesome! Let's investigate on a followup pr, ok?